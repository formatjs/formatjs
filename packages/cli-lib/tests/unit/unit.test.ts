import cliMain from '../../src/cli'
const glob = require('fast-glob')
import ts from 'typescript'
jest.mock('typescript')
// Commander.js will call this.
jest.spyOn(process, 'exit').mockImplementation((() => null) as any)
jest.spyOn(glob, 'sync').mockImplementation(p => (Array.isArray(p) ? p : [p]))

jest.mock('fs-extra', () => ({
  outputJSONSync: () => Promise.resolve(),
  readFile: () => Promise.resolve(';'),
}))

// Since TS5.0 jest mock doesn't seem to work bc of readonly properties
describe.skip('unit', function () {
  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('it passes camelCase-converted arguments to typescript API', async () => {
    await cliMain([
      'node',
      'path/to/formatjs-cli',
      'extract',
      '--extract-source-location',
      '--remove-default-message',
      '--throws',
      '--additional-component-names',
      'Foo,Bar',
      '--ignore=file3.ts',
      'file1.js',
      'file2.tsx',
    ])

    expect((ts.transpileModule as any).mock.calls).toMatchSnapshot()
  })

  it('does not read from stdin when the glob pattern does NOT match anything', async () => {
    // Does not match anything
    jest.spyOn(glob, 'sync').mockImplementation(() => [])
    // This should not hang
    await cliMain(['node', 'path/to/formatjs-cli', 'extract', '*.doesnotexist'])
    expect(ts.transpileModule).not.toHaveBeenCalled()
  }, 500) // 500ms timeout
})
