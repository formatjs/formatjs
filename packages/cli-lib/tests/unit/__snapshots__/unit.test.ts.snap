// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit it passes camelCase-converted arguments to typescript API 1`] = `
[
  [
    ";",
    {
      "compilerOptions": {
        "allowJs": true,
        "experimentalDecorators": true,
        "noEmit": true,
        "target": 99,
      },
      "fileName": "file1.js",
      "reportDiagnostics": true,
      "transformers": {
        "before": [
          [Function],
        ],
      },
    },
  ],
  [
    ";",
    {
      "compilerOptions": {
        "allowJs": true,
        "experimentalDecorators": true,
        "noEmit": true,
        "target": 99,
      },
      "fileName": "file2.tsx",
      "reportDiagnostics": true,
      "transformers": {
        "before": [
          [Function],
        ],
      },
    },
  ],
]
`;
